Error durring update check

Hi,

When I try to check for update, I have the following error in the log:
ERROR - stderr (11:18:59):
Exception in thread Thread-103:
Traceback (most recent call last):
  File "threading.pyc", line 801, in __bootstrap_inner
  File "threading.pyc", line 754, in run
  File "updateCheck.pyc", line 126, in _bg
  File "updateCheck.pyc", line 173, in setNextCheck
  File "wx_misc.pyc", line 1324, in Start
PyAssertionError: C++ assertion "wxThread::IsMain()" failed at ....\src\common\timerimpl.cpp(60) in wxTimerImpl::Start(): timer can only be started from the main thread

I use NVDA version next-13392,621e5393
I might be caused by the recent changes on WX.

Thanks.

I can confirm this.
    Op 7-7-2016 om 11:44 schreef Patrick
      ZAJDA:
    
    
      Hi,
      When I try to check for update, I have the following error in
        the log:
        ERROR - stderr (11:18:59):
        Exception in thread Thread-103:
        Traceback (most recent call last):
        File "threading.pyc", line 801, in __bootstrap_inner
        File "threading.pyc", line 754, in run
        File "updateCheck.pyc", line 126, in _bg
        File "updateCheck.pyc", line 173, in setNextCheck
        File "wx_misc.pyc", line 1324, in Start
        PyAssertionError: C++ assertion "wxThread::IsMain()" failed at
        ....\src\common\timerimpl.cpp(60) in wxTimerImpl::Start(): timer
        can only be started from the main thread
      I use NVDA version next-13392,621e5393
        I might be caused by the recent changes on WX.
      Thanks.
      —
        You are receiving this because you are subscribed to this
        thread.
        Reply to this email directly, view it on
          GitHub, or mute
          the thread.

--- a/source/updateCheck.py
+++ b/source/updateCheck.py
@@ -2,7 +2,7 @@
 #A part of NonVisual Desktop Access (NVDA)
 #This file is covered by the GNU General Public License.
 #See the file COPYING for more details.
-#Copyright (C) 2012-2015 NV Access Limited
+#Copyright (C) 2012-2016 NV Access Limited
 
 """Update checking functionality.
 @note: This module may raise C{RuntimeError} on import if update checking for this build is not supported.
@@ -170,8 +170,9 @@ class AutoUpdateChecker(UpdateChecker):
 		self._checkTimer = None
 
 	def setNextCheck(self, isRetry=False):
-		self._checkTimer.Stop()
-		self._checkTimer.Start((RETRY_INTERVAL if isRetry else CHECK_INTERVAL) * 1000, True)
+		# #6127: Timers must be manipulated from the main thread.
+		wx.CallAfter(self._checkTimer.Stop)
+		wx.CallAfter(self._checkTimer.Start, (RETRY_INTERVAL if isRetry else CHECK_INTERVAL) * 1000, True)
 
 	def _started(self):
 		log.info("Performing automatic update check")
@@ -292,7 +293,8 @@ class UpdateDownloader(object):
 		self._guiExecFunc = func
 		self._guiExecArgs = args
 		if not self._guiExecTimer.IsRunning():
-			self._guiExecTimer.Start(50, True)
+			# #6127: Timers must be manipulated from the main thread.
+			wx.CallAfter(self._guiExecTimer.Start, 50, True)
 
 	def _guiExecNotify(self):
 		self._guiExecFunc(*self._guiExecArgs)

